hr_applicant/models/training.py

```diff
@@ -0,0 +1,310 @@
+        string="Course Type",
+        readonly=True,
+    )
+    training_location = fields.Char(
+        related="course_id.training_location", string="Training Location",
+        readonly=True
+    )
+    training_attendees = fields.Integer("Training Attendees", required=True)
+    training_start_date = fields.Date(
+        "Training Start Date", readonly=True,
+        states={"draft": [("readonly", False)]}, required=True
+    )
+    training_end_date = fields.Date(
+        "Training End Date", readonly=True, states={"draft": [("readonly", False)]}
+    )
+    attendees_ids = fields.One2many(
+        "list.of.attendees", "class_id", string="List of Local Attendees"
+    )
+    state = fields.Selection(
+        [
+            ("draft", "Draft"),
+            ("to_be_approved", "To Be Approved"),
+            ("approved", "Approved"),
+            ("completed", "Completed"),
+            ("cancel", "Cancelled"),
+        ],
+        "State",
+        default="draft",
+    )
+    description = fields.Text("Description")
+
+    @api.onchange("training_start_date", "course_id")
+    def onchange_start_date(self):
+        for rec in self:
+            if rec.training_start_date and rec.course_id:
+                end_date = False
+                training_start_date = (
+                    rec.training_start_date
+                    and datetime.datetime.strftime(
+                        rec.training_start_date, DEFAULT_SERVER_DATE_FORMAT
+                    )
+                )
+                if rec.course_id.duration and rec.course_id.duration_type == "day":
+                    end_date = datetime.datetime.strptime(
+                        training_start_date, DEFAULT_SERVER_DATE_FORMAT
+                    ) + datetime.timedelta(days=rec.course_id.duration - 1)
+                elif rec.course_id.duration and rec.course_id.duration_type == "week":
+                    end_date = datetime.datetime.strptime(
+                        training_start_date, DEFAULT_SERVER_DATE_FORMAT
+                    ) + relativedelta(weeks=rec.course_id.duration, days=-1)
+                elif rec.course_id.duration and rec.course_id.duration_type == "month":
+                    end_date = datetime.datetime.strptime(
+                        training_start_date, DEFAULT_SERVER_DATE_FORMAT
+                    ) + relativedelta(months=rec.course_id.duration, days=-1)
+                end_date = end_date and datetime.datetime.strftime(
+                    end_date, DEFAULT_SERVER_DATE_FORMAT
+                )
+                rec.training_end_date = end_date
+
+    def action_to_be_approve(self):
+        self.write({"state": "to_be_approved"})
+        return True
+
+    def action_approve(self):
+        for rec in self:
+            if not rec.training_attendees:
+                raise ValidationError(
+                    _("Training Attendees should not be Zero!"))
+            rec.write({"state": "approved"})
```

Comment:
We can avoid this type of pattern for assignment rather we can use rec.state="approved".